geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLSpatialAccessorFunctions.scala

```diff
@@ -0,0 +1,87 @@
+package org.apache.spark.sql
+
+import com.vividsolutions.jts.geom._
+
+object SQLSpatialAccessorFunctions {
+  val ST_Boundary: Geometry => Geometry = geom => geom.getBoundary
+  val ST_CoordDim: Geometry => Int = geom => {
+    val coord = geom.getCoordinate
+    if (coord.z.isNaN) 2 else 3
+  }
+  val ST_Dimension: Geometry => Int = geom => geom.getDimension
+  val ST_Envelope: Geometry => Geometry = geom => geom.getEnvelope
+  val ST_ExteriorRing: Geometry => LineString = {
+    case geom: Polygon => geom.getExteriorRing
+    case _ => null
+  }
+  val ST_GeometryN: (Geometry, Int) => Geometry = (geom, n) => geom.getGeometryN(n)
+  val ST_GeometryType: Geometry => String = geom => geom.getGeometryType
+  val ST_InteriorRingN: (Geometry, Int) => Geometry = (geom, n) => {
+    geom match {
+      case geom: Polygon =>
+        if (0 < n && n <= geom.getNumInteriorRing) {
+          geom.getInteriorRingN(n-1)
```

Comment:
do we want to switch to a 1-based offset here? seems like it would make sense to keep it 0 based like we do for st_geometryN. Looking at the postgis docs, they seem to use 1 based for both - but maybe we should not follow that convention.